From b63fb78bf3fcf1daca8cfb3319a2bc0bd4ecbb8f Mon Sep 17 00:00:00 2001
From: Andy Chung <Andy.Chung@amd.com>
Date: Thu, 25 Jun 2026 15:20:26 +0800
Subject: [PATCH] Psusensor: support I3C devices with full 48-bit PID addresses

Psusensor: support I3C devices with full 48-bit PID addresses

PSUSensorMain matches sysfs hwmon devices to Entity Manager config by
parsing "<bus>-<addr>" out of the device path. For i3c devices the addr
component is the full 48-bit I3C PID.

Widen the address regex to {4,12} and parse bus/addr as uint64_t so
I3C devices match get created. I2C devices are unaffected.

Gerrit: https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/91630

Tested:
On the Anacapa BMC, I3C devices can be probed by PSUSensor.

Change-Id: Ibd8ca8ea73b46ca427089c5d52d638fc6675f20b
Signed-off-by: Andy Chung <Andy.Chung@amd.com>
---
 src/psu/PSUSensorMain.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index ba866f7..2b8b169 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -66,7 +66,7 @@
 #include <variant>
 #include <vector>
 
-static std::regex devRegex(R"((\/i[23]c\-\d+\/\d+-[a-fA-F0-9]{4,4})(\/|$))");
+static std::regex devRegex(R"((\/i[23]c\-\d+\/\d+-[a-fA-F0-9]{4,12})(\/|$))");
 static constexpr float gpioBridgeSetupTimeDefault = 1;
 
 static constexpr auto sensorTypes = std::to_array<
@@ -711,8 +711,8 @@ static void createSensorsCallback(
             devType = DevTypes::IIO;
         }
 
-        size_t bus = 0;
-        size_t addr = 0;
+        uint64_t bus = 0;
+        uint64_t addr = 0;
         if (!getDeviceBusAddr(deviceName, bus, addr))
         {
             continue;
-- 
2.53.0-Meta

